hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecordSerializer.java

```diff
@@ -34,49 +36,83 @@
  *
  */
 public class BufferedRecordSerializer<T> implements CustomSerializer<BufferedRecord<T>> {
-  public static final int KRYO_SERIALIZER_INITIAL_BUFFER_SIZE = 1048576;
-  private final Kryo kryo;
-  // Caching ByteArrayOutputStream to avoid recreating it for every operation
-  private final ByteArrayOutputStream baos;
-  private final RecordSerializer<T> recordSerializer;
+  // Caching kryo serializer to avoid creating kryo instance for every serde operation
+  private static final ThreadLocal<InternalSerializerInstance> SERIALIZER_REF =
```

Comment:
The kryo creation happens once for each file group reader so the cost should be low?